test/testdata/resolver/abstract_initialization.rb

```diff
@@ -0,0 +1,27 @@
+# typed: true
+
+class Abstract
+  extend T::Sig
+  extend T::Helpers
+  abstract!
+
+  sig {abstract.void}
+  def foo; end
+end
+
+Abstract.new # error: Attempt to instantiate abstract class `Abstract`
+
+class SubclassOfAbstract < Abstract
+  def foo; end
+end
+
+SubclassOfAbstract.new
+
+class AbstractWithSingletonNew
+  extend T::Helpers
+  abstract!
+
+  def self.new; end
+end
+
+AbstractWithSingletonNew.new
```

Comment:
I think we may want a few more tests here.

https://sorbet.run/#%23%20typed%3A%20true%0A%0A%0Aclass%20Foo%0A%20%20def%20self.new%3B%20end%0Aend%0A%0Aclass%20Bar%20%3C%20Foo%0A%20%20extend%20T%3A%3AHelpers%0A%20%20abstract!%0Aend%0A%0ABar.new%0A

https://sorbet.run/#%23%20typed%3A%20true%0A%0A%0Amodule%20Foo%0A%20%20def%20new%3B%20end%0Aend%0A%0Aclass%20Bar%0A%20%20extend%20Foo%0A%20%20extend%20T%3A%3AHelpers%0A%20%20abstract!%0Aend%0A%0ABar.new%0A

Reply:
We should also add Dmitry's example showing it's not raising any error: 

```rb
class A
  extend T::Helpers
  abstract!
end

class B < A; end

sig{params(a: T.class_of(A)).void}
def takesA(a)
  a.new
end

def assignsA
  a = T.let(B, T.class_of(A))
  a.new 
end
```

Reply:
Added these test cases!